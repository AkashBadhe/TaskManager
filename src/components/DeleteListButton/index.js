import React from "react";
import styled from "styled-components";
import { FaTimesCircle } from "react-icons/fa";
import {
  secondaryBackgroundColor,
  quaternaryBackgroundColor,
} from "../../components/Theme";

const StyledDeleteListButton = styled.button`
  border: 0;
  padding: 0;
  margin-right: 10px;
  background: ${secondaryBackgroundColor};
  color: ${quaternaryBackgroundColor};
  cursor: pointer;
  opacity: 0.8;

  &:hover,
  &:focus,
  &:active {
    color: #da3849;
    opacity: 1;
  }
`;

const DeleteListButton = ({ ...props }) => {
  return (
    <StyledDeleteListButton {...props}>
      <FaTimesCircle size={18} />
    </StyledDeleteListButton>
  );
};

export default DeleteListButton;
