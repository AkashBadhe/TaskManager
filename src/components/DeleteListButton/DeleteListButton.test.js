import React from "react";
import { shallow } from "enzyme";
import DeleteListButton from "./index";

describe("DeleteListButton", () => {
  let deleteListButton = shallow(<DeleteListButton />);
  it("Should mount `DeleteListButton`", () => {
    expect(deleteListButton).toMatchSnapshot();
  });

  it("Should have `styled.button`", () => {
    expect(deleteListButton.find("FaTimesCircle").exists()).toBe(true);
  });
});
