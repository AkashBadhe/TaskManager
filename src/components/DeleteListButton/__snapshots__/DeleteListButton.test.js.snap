// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteListButton Should mount \`DeleteListButton\` 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DeleteListButton />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <FaTimesCircle
        size={18}
/>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "size": 18,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "baseHash": 400283751,
        "componentId": "sc-AxjAm",
        "isStatic": false,
        "rules": Array [
          "
  border: 0;
  padding: 0;
  margin-right: 10px;
  background: #f8f8f8;
  color: rgb(46, 68, 78);
  cursor: pointer;
  opacity: 0.8;

  &:hover,
  &:focus,
  &:active {
    color: #da3849;
    opacity: 1;
  }
",
        ],
        "staticRulesId": "",
      },
      "displayName": "styled.button",
      "foldedComponentIds": Array [],
      "render": [Function],
      "shouldForwardProp": undefined,
      "styledComponentId": "sc-AxjAm",
      "target": "button",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <FaTimesCircle
          size={18}
/>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "size": 18,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "baseHash": 400283751,
          "componentId": "sc-AxjAm",
          "isStatic": false,
          "rules": Array [
            "
  border: 0;
  padding: 0;
  margin-right: 10px;
  background: #f8f8f8;
  color: rgb(46, 68, 78);
  cursor: pointer;
  opacity: 0.8;

  &:hover,
  &:focus,
  &:active {
    color: #da3849;
    opacity: 1;
  }
",
          ],
          "staticRulesId": "",
        },
        "displayName": "styled.button",
        "foldedComponentIds": Array [],
        "render": [Function],
        "shouldForwardProp": undefined,
        "styledComponentId": "sc-AxjAm",
        "target": "button",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
